Update Databricks docs

DimedS opened the pull request:
Closes #5211 

https://kedro--5294.org.readthedocs.build/en/5294/deploy/supported-platforms/databricks/

for test use please last version of the starter, not a `kedro new --tools=pyspark` command:

```
kedro new \
  --starter=https://github.com/kedro-org/kedro-starters.git \
  --directory=spaceflights-pyspark \
  --checkout=main
```

merelcht commented on docs/deploy/supported-platforms/databricks.md:
```diff
@@ -0,0 +1,165 @@
+# Deploying Kedro on Databricks
+
```
I'm missing a bit of an introduction paragraph here explaining what a user can expect from the page. We dive in straight with a `kedro new` command and explanation about the starter, but nothing about the why of Databricks and what this guide will explain.

merelcht commented on docs/deploy/supported-platforms/databricks.md:
```diff
@@ -0,0 +1,165 @@
+
+### Scheduling
+
+- Create a **Databricks Job** that runs the notebook
```
Can we add more explanation on how you can do this? Or at least point people to Databricks docs that explain the process?

merelcht reviewed (commented):
I like that the docs are simpler now, but I feel that the content is a bit sparse. I'm missing a bit more explanation on the _why_ of each option. The page reads like it's just instructions and not a lot of explanation on what happens with each step.

ravi-kumar-pilla commented on docs/deploy/supported-platforms/databricks.md:
```diff
@@ -0,0 +1,165 @@
+
+## Run Kedro *inside* Databricks (Git folders)
+
+This option is suitable if you primarily work **inside the Databricks workspace**, using notebooks and Databricks Jobs. Databricks provides **Git folders**, which allow you to clone a Git repository directly into the workspace and work with it interactively.
```
```suggestion
This option is suitable if you primarily work **within the Databricks workspace**, using notebooks and Databricks Jobs. Databricks provides **Git folders**, which allow you to clone a Git repository directly into the workspace and work with it interactively.
```

ravi-kumar-pilla commented on docs/deploy/supported-platforms/databricks.md:
```diff
@@ -0,0 +1,165 @@
+4. Open the `notebooks/` folder in the cloned repository and create a new notebook.
+5. Attach the notebook to a Databricks cluster (for example, a serverless cluster).
+
+### Run Kedro from a notebook
```
The above heading - Run Kedro *inside* Databricks (Git folders) and Run Kedro from a notebook seems to be identical though Run Kedro from a notebook is a sub-heading. Do we need this sub-heading ?

ravi-kumar-pilla commented on docs/deploy/supported-platforms/databricks.md:
````diff
@@ -0,0 +1,165 @@
+```ipython
+%run_viz
+```
+
````
Can we also add a section to include [NotebookVisualizer](https://docs.kedro.org/projects/kedro-viz/en/stable/kedro-viz_visualisation/#notebookvisualizer) ? Thank you

rashidakanchwala commented on docs/deploy/supported-platforms/databricks.md:
````diff
@@ -0,0 +1,165 @@
+## Create a Spark-enabled Kedro project
+
+```bash
+uvx kedro new --name=my-project --tools=spark --example=y
````
```suggestion
uvx kedro new --name=my-project --tools=pyspark --example=y
```

DimedS commented:
Many thanks for your reviews and comments, @merelcht, @SajidAlamQB, @ravi-kumar-pilla, and @rashidakanchwala!
I’ve tried to address all of them: I added an explanation and a comparison table at the top, reworked the document to be more tutorial-style for users less familiar with Databricks, included more explanations with links to the relevant Databricks docs, and revised the notebook execution section to use line magics. I also added NotebookVisualizer.

I’ve requested a re-review and would appreciate it if you could give it one more pass.

merelcht reviewed (commented):
Thanks Dmitry, I think this is going into the right direction! To me it does feel like the docs are targeted to a more advanced audience than before and has some assumptions that a user is able to setup and check things work independently. I think it would be good to add some more code snippets and/or screenshots similar to the airflow ad prefect docs as guidance.

ElenaKhaustova commented on docs/deploy/supported-platforms/databricks.md:
```diff
@@ -0,0 +1,221 @@
+
+### Prerequisites
+
+Make sure Kedro and Kedro-Viz are installed in the same scope(notebook-scoped or cluster-scoped). For notebook-scoped installs:
```
```suggestion
Make sure Kedro and Kedro-Viz are installed in the same scope (notebook-scoped or cluster-scoped). For notebook-scoped installs:
```

ElenaKhaustova reviewed (approved):
Thank you, @DimedS, looks great 👍

rashidakanchwala reviewed (approved):
With Databricks Free Edition, I was able to follow most of the tutorial successfully.

What I tested:
- Running session.run() from a notebook inside the Kedro repo
- Running session.run() from a notebook outside the repo
→ The Notebook Visualiser worked in both cases.

What didn’t work:
- %run_viz ran for ~5 minutes with no output and never completed; I eventually had to interrupt it.

It would be good if someone else could try and confirm this, as I believe it works for Dmitry. If `%run_viz` turns out to be an issue, I don’t think it should block this PR — we can always raise a separate PR to address it.

Last suggested change for this PR:
- Make Notebook Visualiser the default / first way of viewing pipelines in a Databricks notebook.
- Mention %run_viz as an option for users who want the full Kedro-Viz browser experience, rather than positioning it as the primary path.

Things I haven’t tested because of free Databricks limitation but the docs look great:
- databricks-connect 
- kedro-databricks plugin

rashidakanchwala commented:
Awesome work @DimedS and @SajidAlamQB !!

merelcht reviewed (approved):
Thanks for addressing my comments @DimedS, this is much easier to follow now 👍 
Left one final suggestion, but otherwise this looks all good to me.
